api/apps/api/src/modules/scenarios-features/scenario-features.service.ts

```diff
@@ -79,6 +79,14 @@ export class ScenarioFeaturesService extends AppBaseService<
     ];
   }
 
+  async isFeaturePresentInAnyScenario(featureId: string): Promise<boolean> {
+    const count = await this.remoteScenarioFeatures.query(
+      `select count(*) from scenario_features_data where feature_class_id in (select id from features_data where feature_id = '${featureId}')`,
+    );
+
+    return count > 0;
```

Comment:
secondly, although in theory the query above looks legit (_BUT_ I need some time to think through the implications of split features on this query, which I can only do tomorrow morning unfortunately), in practice you end up boycotting the query result in this check here, because `count` is not what you think it will be, and `count > 0` will always be `true`

If you inspect this, the return of `query()` will be something like this:

```
[ { count: '3' } ]
```

As a corollary of this, I think you should also review the test that checks that deletion of in-use features is prevented, as this bug should be caught there, and if the test passes then... I think the test it's not correct.

Try creating a project/scenario, uploading some features (fun time with bots?!) and link them to a scenario, and then delete these via the new API endpoint.